src/target/ext/memory_map.rs

```diff
@@ -0,0 +1,14 @@
+//! Provide a memory map for the target.
+use crate::target::Target;
+
+/// Target Extension - Provide a target memory map.
+pub trait MemoryMap: Target {
+    /// Return the target memory map XML file.
+    ///
+    /// See the [GDB Documentation] for a description of the format.
+    ///
+    /// [GDB Documentation]: https://sourceware.org/gdb/onlinedocs/gdb/Memory-Map-Format.html
+    fn memory_map_xml(&self) -> String;
```

Comment:
I think the CI's already caught this, but I'll add a comment as well.

`gdbstub` is a `no_std`, no `alloc` project, which means `String` isn't something we can use as part of the public API. Instead, this method ought to return a `&str` instead.

Note that this doesn't preclude the implementor from dynamically constructing the memory map string themselves. e.g:

```rust
struct MyTarget {
    memory_map_xml: Option<String>,
}

impl MemoryMap for MyTarget {
    fn memory_map_xml(&self) -> &str {
        match self.memory_map_xml {
            Some(xml) => &xml, // return buffered map
            None => {
                // dynamically construct memory map, assigning to `self.memory_map_xml`
                self.memory_map_xml.as_deref().unwrap()
            }
        }
    }
}
```